TFT/src/User/API/MachineParameters.h

```diff
@@ -10,29 +10,29 @@ extern "C" {
 
 typedef enum
 {
-  P_STEPS_PER_MM = 0,
+  P_HOME_OFFSET = 0,
+  P_PROBE_OFFSET,
+  P_MBL_OFFSET,
+  P_OFFSET_TOOL,
+  P_ABL_STATE,
+  P_STEPS_PER_MM,
   P_CURRENT,
   P_MAX_FEED_RATE,
   P_MAX_ACCELERATION,
   P_ACCELERATION,
   P_JERK,
   P_JUNCTION_DEVIATION,
-  P_PROBE_OFFSET,
-  P_HOME_OFFSET,
   P_BUMPSENSITIVITY,
   P_FWRETRACT,
   P_FWRECOVER,
   P_AUTO_RETRACT,
   P_LIN_ADV,
-  P_ABL_STATE,
-  P_OFFSET_TOOL,
   P_HYBRID_THRESHOLD,
   // Keep below items always at the end
   P_SAVE_SETTINGS,
   P_RESTORE_SETTINGS,
   P_RESET_SETTINGS,
-  P_ITEMSCOUNT,
-  P_MBL_Z_OFFSET
+  P_ITEMSCOUNT
 } PARAMETER_NAME;
```

Comment:
the goal should be to keep less common settings at the end and maintain the sequence as in M503 reporting and in the marlin firmware's settings menu.

Reply:
yes, I agree. The offset / ABL params are also used in different menus (Z offset menu, BL menu etc...), so frequently used/changed. For that reason I ordered (the offset params) and moved some those few params on top of the list.
Also, in the language file, I grouped and ordered all the parameters (machine param and config.ini param) in the same order they are provided in config.ini / ParameterSettings menu. In FeatureSettings menu I also show the items in the same order as in config.ini. It's more clear to navigate and see the param in the same order as they appear in config.ini. Also it make clear and esay to provide new param in the language files. I hope you agree on that.

Reply:
I do not have any issue in reorganizing the language keywords. The parameter settings should be arranged according to the sequence in marlin and the most basic parameter should be on top. like for eg. majority of the machines are set up with ABL, putting P_MBL_OFFSET on top is not good. Even in the configuration.h file, the offsets, leveling points, etc are placed after the basic settings. Home offsets are set only once during the first setup of the machine, putting it on top is not good logic. The arrangement should be consistent with that of marlin firmware. In the same way in config files, most common settings are placed on top followed by less and less common/device-specific settings.

Reply:
I just arranged some params (offset tool, current, bump sensitivity) in ParameterSettings according to Marlin (sequence in ParseAck). But it's not clear to me which param you consider basic. The offset and ABL params are changed more frequently than the other machine settings such as steps, acceleratation etc (those for me were never changed because already properly set on Marlin fw).
Currently, only home, probe and mbl offset were put on top just because they seems to be the only parameters that you can change more frequently (the offset normally needs to be changed according to the env temperature).
It'is not a problem for me to follow exacly the same sequence as in Marlin but doing this it seems to me that in that case your suggestion (provide on top the most frequently used params) is disowned

Reply:
in parseACK the parameters are arranged according to the gcode sequence numbers. When some parameters like ABL state are already available in other menus then why put them in the top? Why do homes offsets need to be changed so frequently? it is only the z offset which may change, but it also has its own menu. then why put it on top everywhere? the probe offsets also does not change after setup..

Reply:
Home and Probe offset in ParameterSettings allow also to change XY. Those offset are changed more frequently than any other setting in that menu.
A good optimization on parseAck would be to move on top of the check list the ack messages instead of the machine and capabilities that usually are triggered only at startup or to refresh the machine params.
The ack messages are typically exchanged during a print so when the system is stressed. Moving on top tha ack messages handling will improve the responsivness